packages/sodium/install

```diff
@@ -2,8 +2,27 @@
 set -x
 set -e
 
+OS_DISTRIBUTION=$(lsb_release -cs)
+
 apt-key adv --keyserver keyserver.ubuntu.com --recv-keys C7917B12
-echo "deb http://ppa.launchpad.net/chris-lea/libsodium/ubuntu $(lsb_release -cs) main" >> /etc/apt/sources.list.d/libsodium-ppa.list
+
+function add_repo {
+    echo "deb http://ppa.launchpad.net/chris-lea/libsodium/ubuntu $OS_DISTRIBUTION main" >> /etc/apt/sources.list.d/libsodium-ppa.list
+}
+
+# check for supported platform (whitelisted ubuntu versions)
+if [ $OS_DISTRIBUTION == "precise" ]; then
+    add_repo
+elif [ $OS_DISTRIBUTION == "trusty" ]; then
+    add_repo
+elif [ $OS_DISTRIBUTION == "vivid" ]; then
+    add_repo
+else
+    echo "This platform is unsupported ($OS_DISTRIBUTION) by the apt repository. Skipping"
+    test -f /etc/apt/sources.list.d/libsodium-ppa.list && rm /etc/apt/sources.list.d/libsodium-ppa.list
```

Comment:
Why remove an existing sources.list.d entry? Some other actor may know better than this script.

Also: No need to `test -f`; just `rm -f`, which is fine if files do not exist.

Reply:
Im removing the file because we've already added the file on several of our hosts (which was wrong to do), so this fixes that. But I could also remove those files manually if you'd prefer

the `test -f` is needed because `rm` will exit 1 which cause the script to fail (due to `set` at the top of the script). Even if that wasn't the case, i'd prefer not to print out an error that doesn't matter.

Reply:
I'd rather see a manual cleanup. Until all the installers are vetted against Xenial, YMMV and cruft may appear. Each installer should not then be updated to remove the cruft.